services/provider/server/server.go

```diff
@@ -405,6 +419,21 @@ func (s *OCSProviderServer) GetDesiredClientState(ctx context.Context, req *pb.G
 			return nil, status.Errorf(codes.Internal, "failed to produce client state hash")
 		}
 
+		isRbdDriverEnabled := consumerConfig.GetRbdClientProfileName() != ""
+		if isRbdDriverEnabled {
+			response.RbdDriverRequirements = &pb.RbdDriverRequirements{}
+		}
+
+		isCephFsDriverEnabled := consumerConfig.GetCephFsClientProfileName() != ""
+		if isCephFsDriverEnabled {
+			response.CephFsDriverRequirements = &pb.CephFsDriverRequirements{}
+		}
+
+		isNfsDriverEnabled := consumerConfig.GetNfsClientProfileName() != ""
+		if isNfsDriverEnabled {
+			response.NfsDriverRequirements = &pb.NfsDriverRequirements{}
+		}
```

Comment:
@rohan47 I am not sure this is the correct source of truth for the decision you are trying to make? What is the thinking behind basing the decision on profile name existence in the consumers configmap?

Reply:
I was using the Profile names because we were enabling drivers for a client for a particular storageconsumer, and consumerConfig only gives us profilenames.

Changed the implementation to user `util.GetAvailableServices()`

Reply:
@nb-ohad using `util.GetAvailableServices()` gets all the running services based on the StorageCluster and not StorageConsumer, and also requires ocs-provider-server to have permissions to get cephfilesystems, cephnfses and noobaas.

It will make sense to use distributed storageClasses as the source of truth for enabling drivers.
If you agree with using storageClasses I will make the changes to this PR.

cc @leelavg